tooling/hcpctl/pkg/breakglass/minting/minting.go

```diff
@@ -0,0 +1,233 @@
+	}
+
+	scheme, err := common.NewCertificatesScheme()
+	if err != nil {
+		return nil, err
+	}
+
+	ctrlClient, err := client.New(restConfig, client.Options{Scheme: scheme})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create controller client: %w", err)
+	}
+
+	return &DefaultManager{
+		kubeClient: kubeClient,
+		ctrlClient: ctrlClient,
+	}, nil
+}
+
+// generateCSRName generates a unique CSR name
+func (mgr *DefaultManager) generateCSRName(user string) string {
+	return fmt.Sprintf("%s-%s-%d", defaultCSRNamePrefix, user, time.Now().Unix())
+}
+
+// generateSignerName generates the signer name for the CSR
+func (mgr *DefaultManager) generateSignerName(namespace string) string {
+	return fmt.Sprintf(defaultSignerNameTemplate, namespace)
+}
+
+// generateLabels generates labels for the CSR and approval resources
+func (mgr *DefaultManager) generateLabels(clusterID, sanitizedUser string) map[string]string {
+	return map[string]string{
+		"api.openshift.com/id":   clusterID,
+		"api.openshift.com/name": sanitizedUser,
+		"api.openshift.com/type": "break-glass-credential",
+	}
+}
+
+// CreateCSR creates a new Certificate Signing Request using the provided parameters.
+func (mgr *DefaultManager) CreateCSR(ctx context.Context, csrPEM []byte, clusterID, user, namespace string) (string, error) {
+	sanitizedUser, err := utils.SanitizeUsername(user)
+	if err != nil {
+		return "", fmt.Errorf("invalid username for CSR naming: %w", err)
+	}
+
+	csrName := mgr.generateCSRName(sanitizedUser)
+	signerName := mgr.generateSignerName(namespace)
+	labels := mgr.generateLabels(clusterID, sanitizedUser)
+
+	csr := &certificatesv1.CertificateSigningRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   csrName,
+			Labels: labels,
+		},
+		Spec: certificatesv1.CertificateSigningRequestSpec{
+			Request:           csrPEM,
+			SignerName:        signerName,
+			ExpirationSeconds: func() *int32 { v := defaultExpirationSeconds; return &v }(),
+			Usages: []certificatesv1.KeyUsage{
+				certificatesv1.UsageClientAuth,
+				certificatesv1.UsageDigitalSignature,
+			},
+		},
+	}
+
+	_, err = mgr.kubeClient.CertificatesV1().CertificateSigningRequests().Create(ctx, csr, metav1.CreateOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to create CSR: %w", err)
+	}
+
+	return csrName, nil
+}
+
+// CreateCSRApproval creates an approval resource for the CSR using the provided parameters.
+func (mgr *DefaultManager) CreateCSRApproval(ctx context.Context, csrName, namespace, clusterID, user string) error {
+	sanitizedUser, err := utils.SanitizeUsername(user)
+	if err != nil {
+		return fmt.Errorf("invalid username for kubeconfig authInfo: %w", err)
+	}
+	labels := mgr.generateLabels(clusterID, sanitizedUser)
+
+	approval := &certificatesv1alpha1.CertificateSigningRequestApproval{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      csrName,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+	}
+
+	if err := mgr.ctrlClient.Create(ctx, approval); err != nil {
```

Comment:
We should have typed clients generated for these - if you can, prefer to use them over controller runtime, lots of bizarre things happen in controller-runtime codebase that you don't want to deal with.